src/mixtures/mixturemodel.jl

```diff
@@ -477,6 +477,22 @@ rand(rng::AbstractRNG, s::MixtureSampler{Univariate}) =
 rand(rng::AbstractRNG, d::MixtureModel{Univariate}) =
     rand(rng, component(d, rand(rng, d.prior)))
 
+function rand(rng::AbstractRNG, d::MixtureModel{Univariate}, n::Int)
+    counts = rand(rng, Multinomial(n, probs(d.prior)))
+    x = Vector{partype(d)}(undef, n)
```

Comment:
`partype` is not correct in general either. The only safe way is to perform sampling.